src/forder.c

```diff
@@ -445,11 +445,11 @@ SEXP forder(SEXP DT, SEXP by, SEXP retGrpArg, SEXP sortGroupsArg, SEXP ascArg, S
       Rprintf("forder.c received %d rows and %d columns\n", length(VECTOR_ELT(DT,0)), length(DT));
   }
   if (!length(DT))
-    error("DT is an empty list() of 0 columns");
+    error("Internal error: DT is an empty list() of 0 columns");  // # nocov  should have been caught be colnamesInt, test 2099.1
```

Comment:
does it make sense to push down call to `colnamesInt` from R to C? more related logic next to each other, and (probably very tiny) speed improvement.

Reply:
colnamesInt is already a .Call to C. You added it quite recently iirc and it's nice. There are 11 calls currently from R level which seems right and nice to me.

Reply:
I mean to call it from forder.c not from forder R

Reply:
Could do I suppose.  I've just made forderv even smaller at R level, so forderv could be a single .Call() eventually.  Moving `colnamesInt` call down to C for that reason might be nice. I feel like I've spent way too long on this already so let's pencil that in for future, if that's ok.